packages/ui-components/src/components/layout/PageTabs.vue

```diff
@@ -101,38 +177,82 @@ const activeItemRef = computed(() => {
   const parent = buttonContainer.value
   if (!parent) return null
 
-  const btns = [...parent.getElementsByClassName('tab-button')] as HTMLButtonElement[]
+  const btns = [...parent.getElementsByClassName('tab-button')] as HTMLElement[]
   return btns.find((b) => b.dataset['tabId'] === id) || null
 })
 
-const borderStyle = computed(() => {
+const borderStyle = computed<CSSProperties>(() => {
   const element = activeItemRef.value
-  const style: CSSProperties = {
+  return {
     left: `${element?.offsetLeft || 0}px`,
     width: `${element?.clientWidth || 0}px`
   }
-  return style
 })
 
 const setActiveItem = (item: LayoutPageTabItem) => {
   activeItem.value = item
+  isInitialSetup.value = false
+}
+
+const isActiveItem = (item: LayoutPageTabItem) => {
+  return activeItem.value?.id === item.id
+}
+
+const checkArrowsVisibility = () => {
+  const container = scrollContainer.value
+  if (!container) return
+
+  const scrollWidth = container.scrollWidth
+  const clientWidth = container.clientWidth
+  const scrollLeft = container.scrollLeft
+  const buffer = 1
+
+  showLeftArrow.value = scrollLeft > buffer
+  showRightArrow.value = scrollLeft < scrollWidth - clientWidth - buffer
 }
 
-if (isClient) {
-  // Doing onMounted & watch separately to avoid hydration mismatch
-  onMounted(() => {
+const scrollLeft = () => {
+  scrollContainer.value?.scrollBy({ left: -100, behavior: 'smooth' }) // Adjust the scroll amount as needed
+  checkArrowsVisibility()
+}
+
+const scrollRight = () => {
+  scrollContainer.value?.scrollBy({ left: 100, behavior: 'smooth' }) // Adjust the scroll amount as needed
+  checkArrowsVisibility()
+}
+
+const handleScroll = () => {
+  checkArrowsVisibility()
+}
+
+onMounted(() => {
+  if (isClient) {
     if (props.items.length && !activeItem.value) {
       setActiveItem(props.items[0])
     }
-  })
-
-  watch(
-    () => [props.items, activeItem.value] as const,
-    ([newItems, activeItem]) => {
-      if (newItems.length && !activeItem) {
-        setActiveItem(newItems[0])
-      }
+    checkArrowsVisibility()
+  }
+})
+
+watch(
+  () => [props.items, activeItem.value] as const,
+  ([newItems]) => {
+    if (Array.isArray(newItems) && newItems.length && !activeItem.value) {
+      setActiveItem(newItems[0] as LayoutPageTabItem)
     }
-  )
-}
+    checkArrowsVisibility()
+  }
+)
 </script>
+<style>
+/* Hide scrollbar for Chrome, Safari and Opera */
+.hide-scrollbar::-webkit-scrollbar {
```

Comment:
Weird, why not just disable overflow? And if we do in fact want scrolling, then surely hiding the scrollbar is bad accessibility?

Reply:
We don't want to disable overflow, as it does still scroll. The issue with the scrollbar is that it looked ugly sitting below the menu items. It also obscured the sliding underline which made it hard to see the active item. 

As an alternative to the scrollbar, I added arrows and a fade to each side of the menu, when there was scroll available:

https://github.com/specklesystems/speckle-server/assets/139135120/e83fe5ea-3612-4562-91d1-884b21347870

This is based on how Dribble handle this problem, which Benjamin suggested.

Reply:
@andrewwallacespeckle Ok, good enough. But please mention this as a comment, cause other devs might have the same question